### PR TITLE
Expose pipeline component

### LINKED ISSUE #172: Expose `PipelineComponent` in defaults
We can expose `PipelineComponent` in the defaults and use some properties like `prefix` to set global defaults for all the components.
```yaml
pipeline-component:
  prefix: ""
```

### PR DESCRIPTION
**Pull request by irux**
closes #172

### CODE REVIEW

**Review by raminqaf — CHANGES_REQUESTED** [review]
Could you please add some (snapshot) tests?

**Review by disrupted — CHANGES_REQUESTED** [review]
the pipeline schema needs to be updated

File: kpops/components/base_components/pipeline_component.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 class PipelineComponent(BaseDefaultsComponent):
     name: str
+    type: str = "pipeline-component"
```
Comment (disrupted) [inline kpops/components/base_components/pipeline_component.py:26]: type should be the first attribute

**Review by disrupted — CHANGES_REQUESTED** [review]

**Comment by irux** [comment]
> the pipeline schema needs to be updated

what should I update?

**Comment by sujuka99** [comment]
Should we also include `PipelineComponent` in the pipeline schema?

If yes, then I believe a `schema_type` attr is also required.

File: kpops/components/base_components/pipeline_component.py [hunk 2]
```diff
@@ -30,6 +32,9 @@ class PipelineComponent(BaseDefaultsComponent):
         default="${pipeline_name}-",
         description="Pipeline prefix that will prefix every component name. If you wish to not have any prefix you can specify an empty string.",
     )
+    schema_type: Literal["pipeline-component"] = Field(  # type: ignore[assignment]
+        default="pipeline-component", exclude=True
+    )
```
Comment (disrupted) [inline kpops/components/base_components/pipeline_component.py:37]: please move `schema_type` directly underneath the `type` attribute as it is the case for all the other component classes

File: tests/pipeline/resources/pipeline-component-should-have-prefix/defaults.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+pipeline-component:
+  prefix: "from-pipeline-component-"
+
+kubernetes-app:
+  namespace: ${NAMESPACE}
+
+kafka-connector:
+  namespace: ${NAMESPACE}
```
Comment (disrupted) [inline tests/pipeline/resources/pipeline-component-should-have-prefix/defaults.yaml:8]: what is this used for?

File: tests/pipeline/resources/pipeline-component-should-have-prefix/pipeline.yaml
Comment (disrupted) [inline tests/pipeline/resources/pipeline-component-should-have-prefix/pipeline.yaml:1]: please fix the end of the file (missing newline)

**Review by disrupted — CHANGES_REQUESTED** [review]

File: tests/pipeline/resources/pipeline-component-should-have-prefix/defaults.yaml (hunk 3 above)
Comment (irux) [reply tests/pipeline/resources/pipeline-component-should-have-prefix/defaults.yaml:8]: you mean why is this as a env variable or why is this declared ?

**Review by disrupted — APPROVED** [review]
LGTM

**Review by sujuka99 — DISMISSED** [review]
LGTM

**Review by raminqaf — APPROVED** [review]

**Pull request merged**